lib/propmangle.js

```diff
@@ -101,7 +105,8 @@ function mangle_properties(ast, options) {
         }
         else if (node instanceof AST_Sub) {
             if (this.parent() instanceof AST_Assign) {
-                addStrings(node.property);
+                if (!(ignore_quoted && node.property.quote))
+                    addStrings(node.property);
             }
         }
     }));
```

Comment:
You also have to put similar logic in the next section (step 2: transform the tree, renaming properties) otherwise the following will not be transformed correctly:
```
        function f1() {
            a["foo"] = "bar";
            a.color = "red";
            x = {"bar": 10};
        }
        function f2() {
            a.foo = "bar";
            a['color'] = "red";
            x = {bar: 10};
        }
```